src/data/model.ts

```diff
@@ -0,0 +1,35 @@
+export default class Model {
+  private translateFields = {};
+
+  public factory(data: any): any {
+    for (let key in data) {
+      if (data.hasOwnProperty(key)) {
+        let rawData = data[key];
+        key = this.translate(key);
+        if (!this.hasOwnProperty(key) || typeof rawData === "undefined") {
+          continue;
+        }
+        let goDeep = this[key] instanceof Model;
+        if (goDeep) {
+          this[key].factory(rawData);
+          continue;
+        }
+        (<any>this)[key] = rawData;
+      }
+    }
+  }
+
+  public translate(key: string) {
+    if (!this.translateFields) {
+      return key;
+    }
+    if (!this.translateFields[key]) {
```

Comment:
You can combine this with the above condition and check if the `typeof  this.translateFields[key] === 'undefined' `because the translation can be 0 or false I don't know this is not the right approach to check if exists